Dockerfile

```diff
@@ -1,17 +1,94 @@
-#
-# To build this Docker image: docker build -t fpm .
-#
-# To run this Docker container interactively: docker run -it fpm
-#
-FROM alpine:3.12
-
-RUN apk add --no-cache \
-        ruby \
-        ruby-dev \
-        ruby-etc \
-        gcc \
-        libffi-dev \
-        make \
-        libc-dev \
-        rpm \
-    && gem install --no-document fpm
+# Are we running against the minimal container, or the everything
+# container? Minimal is mostly the compiled package tools. Everything
+# pulls in scripting langauges.
+ARG BASE_ENV=everything
+
+# Are we running tests, or a release? Tests build and run against the
+# CWD, where release will use the downloaded gem.
+ARG TARGET=test
+
+# Container to throw an error if called with a bare `docker build .`
+FROM ubuntu:18.04 as error
+RUN echo "\n\n\nHey! Use buildkit. See the Makefile or docs\n\n\n"
+RUN false
+
+# Base container is used for various release and test things
+FROM ubuntu:18.04 as minimal-base
+
+# Runtime deps. Build deps go in the build or test containers
+RUN apt-get update \
+	&& apt-get -y dist-upgrade \
```

Comment:
which of these are required to package RPM?

Reply:
at least `rpm` because it provides `rpmbuild`
https://github.com/jordansissel/fpm/blob/d819e59a806e5b8bdb52209d7245e14d32d97cf9/lib/fpm/package/rpm.rb#L447

Reply:
verified all I was missing from the ubuntu image was RPM.  Leaving the comment for others.